mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp

```diff
@@ -1803,64 +1822,119 @@ void SiconosBulletCollisionManager_impl::createCollisionObject(
   // btConvexHullShape* childShape1 = new btConvexHullShape(&pts[0],rows, sizeof(btScalar)*3);
 
   btConvexHullShape * btch;
-  btch = new btConvexHullShape(&pts[0], rows, sizeof(btScalar)*3);  // Warning: Possible loss of memory since we cannot SP here
+  btch = new btConvexHullShape(&pts[0], rows, sizeof(btScalar)*3);  // Warning: Possible loss of memory
```

Comment:
It depends on what removeCollsionObject do on the bullet side, btConvex2d does not free its parameter. A  simple solution here is to create the pointer as a smart pointer (SP::btCOnvexHullShape is defined) and attach this smart pointer into a structure which has the same lifetime as the object (a bit like bodyShapeRecord).